cupy/misc/who.py

```diff
@@ -0,0 +1,94 @@
+import sys
+import cupy as cp
+
+
+def who(vardict=None):
+
+    """
+    Print the CuPy arrays in the given dictionary.
+    If there is no dictionary passed in or `vardict` is None then returns
+    CuPy arrays in the globals() dictionary (all CuPy arrays in the
+    namespace).
+    Parameters
+    ----------
+    vardict : dict, optional
+        A dictionary possibly containing ndarrays.  Default is globals().
+    Returns
+    -------
+    out : None
+        Returns 'None'.
+    Notes
+    -----
+    Prints out the name, shape, bytes and type of all of the ndarrays
+    present in `vardict`.
+    Examples
+    --------
+    >>> a = cp.arange(10)
+    >>> b = cp.ones(20)
+    >>> cp.who()
+    Name            Shape            Bytes            Type
+    ===========================================================
+    a               10               80               int64
+    b               20               160              float64
+    Upper bound on total bytes  =       240
+    >>> d = {'x': cp.arange(2.0), 'y': cp.arange(3.0), 'txt': 'Some str',
+    ... 'idx':5}
+    >>> cp.who(d)
+    Name            Shape            Bytes            Type
+    ===========================================================
+    x               2                16               float64
+    y               3                24               float64
+    Upper bound on total bytes  =       40
+    """
+
+    if vardict is None:
+        frame = sys._getframe().f_back
```

Comment:
why not `globals()`?

Reply:
Wouldn't that return the stack the function is in, not the stack it was called from?

Reply:
Good point